src/algebra/big_operators/fin.lean

```diff
@@ -161,6 +161,20 @@ end partial_prod
 
 end fin
 
+namespace vector
+
+lemma sum_ite_eq_nth_eq_count [decidable_eq α] [add_comm_monoid_with_one β] {n : ℕ}
+  (a : α) (v : vector α n) : ∑ i, ite (a = v.nth i) (1 : β) 0 = ↑(v.to_list.count a) :=
```

Comment:
This lemma still feels weird to me; I wonder if there's a more general lemma we can extract from it?

I think `rw ←finset.sum_filter` takes this to the more pleasant
```suggestion
  (a : α) (v : vector α n) : ∑ i in univ.filter (λ i, a = v.nth i), 1 = ↑(v.to_list.count a) :=
```
From which I think the actually interesting statement is that `(univ.filter (λ i, a = v.nth i)).count = v.to_list.count`

Reply:
I made this change and added a proof for the second statement. Not sure about any other generalizations.